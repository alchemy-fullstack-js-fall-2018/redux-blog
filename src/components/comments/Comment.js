import React from 'react';
import PropTypes from 'prop-types';

function Comment({ comment }) {
  const { name, email, body } = comment;

  return (
    <li>{name} - {email}
      <p>{body}</p>
    </li>
  );
}

Comment.propTypes = {
  comment: PropTypes.object.isRequired
};

export default Comment;
