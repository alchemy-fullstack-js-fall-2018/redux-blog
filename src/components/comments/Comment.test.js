import React from 'react';
import { shallow } from 'enzyme';
import Comment from './Comment';

describe('Comment snapshot', () => {
  it('matches previous snapshot', () => {
    const comment = {
      postId: 1,
      id: 1,
      name: 'id labore ex et quam laborum',
      email: 'Eliseo@gardner.biz',
      body:
        'laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium'
    };

    const wrapper = shallow(<Comment comment={comment} />);
    expect(wrapper).toMatchSnapshot();
  });
});
