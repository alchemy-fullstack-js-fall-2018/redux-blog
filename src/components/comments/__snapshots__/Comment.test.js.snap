// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment snapshot matches previous snapshot 1`] = `
<Fragment>
  <h5>
    id labore ex et quam laborum
    : 
    Eliseo@gardner.biz
  </h5>
  <p>
    laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
  </p>
  <br />
</Fragment>
`;
